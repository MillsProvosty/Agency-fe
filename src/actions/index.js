export const setRole = role => ({
  type: 'SET_ROLE',
  role
});

export const setUser = user => ({
  type: 'SET_USER',
  user
});

export const setUserOpportunities = opportunities => ({
  type: 'SET_USER_OPPORTUNITIES',
  opportunities
});

export const addOpp = opp => ({
  type: 'ADD_OPP',
  opp
});

export const editOpp = opp => ({
  type: 'EDIT_OPP',
  opp
});
