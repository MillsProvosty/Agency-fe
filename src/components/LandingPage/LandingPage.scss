.main {
    margin: 100px;
    height: auto;
  & h1 {
    font-size: 6em;
    margin: 0;
  }

  & p {
    font-size: 2em;
    margin: 2px;
  }

  & section {
    justify-content: space-around;

    & button {
      color: white;
      background-color: darkblue;
      border-radius: 5px;
      font-size: 2em;
      width: 175px;
      font-family: "Quicksand", sans-serif;
      border: 2px solid white;
    }

    & button:hover {
      border: 2px solid darkblue;
      color: darkblue;
      background-color: white;
    }

    & .client {
      margin-left: 20px;
      margin-top: 100px;
    }

    & .volunteer {
      margin-top: 100px;
      margin-right: 20px;
    }
  }
}
@media screen and (max-width: 375px) {
  .main {
    & h1 {
      margin: 0px 0px 30px -80px
    }
    & section {
      justify-content: space-around;
  
      & .client {
        margin: 20px 0px 0px 0px;
      }
  
      & .volunteer {
        margin-top: 50px;
        margin-right: 20px;
      }
    }
  }
}

@media all and (display-mode: standalone) {
  .main {
    & h1 {
      margin: 0px 0px 30px -80px
    }
    & section {
      justify-content: space-around;
  
      & .client {
        margin: 20px 0px 0px 0px;
      }
  
      & .volunteer {
        margin-top: 50px;
        margin-right: 20px;
      }
    }
  }
}