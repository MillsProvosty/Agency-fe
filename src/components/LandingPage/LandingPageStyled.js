import styled from "styled-components";
import balloonImg from "./air_support.svg";

  export const Button = styled.button`
    color: white;
    background-color: #37474e;
    border-radius: 5px;
    font-size: 2em;
    width: 175px;
    font-family: "Quicksand", sans-serif;
    border: 2px solid white;
    margin-top: 30px;
    margin-right: ${props => (props.volunteer ? "20px" : "0px")};
    margin-left: ${props => (props.client ? "20px" : "0px")} @media screen and
      (max-width: 375px) {
      margin: ${props =>
        props.client ? "20px 0px 0px 0px" : "50px 20px 0px 0px"};
    }

    :hover {
      border: 2px solid #37474e;
      color: #37474e;
      background-color: white;
    }
  `;

  export const PTag = styled.p`
    font-size: 2em;
    margin: 2px;
  `;

  export const Header = styled.h1`
    color: #37474E
    font-size: 6em;
    margin: 0px;
    @media screen and (max-width: 375px) {
      margin: 0px 0px 30px -80px
    }
  `;

  export const Section = styled.section`
    height: 100vh;
    width: 100%;
    background-color: aliceblue;
    background-repeat: no-repeat;
    background-size: 1000px;
    background-position: top center;
    background-image: url(${balloonImg});
    @media screen and (max-width: 375px) {
      justify-content: space-around;
    }
  `;

  export const Logo = styled.button`
  margin-top:100px;
  border-radius: 50%;
  border: 3px solid #37474e;
  `
