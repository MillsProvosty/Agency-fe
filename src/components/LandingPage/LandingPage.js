import React, { useState } from "react";
import "./LandingPage.scss";
import { NavLink } from "react-router-dom";
import { GiAirBalloon } from "react-icons/gi";
import { connect } from "react-redux";
import { setRole } from "../../actions";
import { Button, PTag, Header, Section, Logo } from "./LandingPageStyled";

export const LandingPage = props => {
<<<<<<< HEAD


  const [userRole, setTheRole] = useState(false)
  console.log('landing', props)
=======
  const [userRole, setTheRole] = useState(false);
  console.log("landing", props);
>>>>>>> f7999fff44d6f0a4964f1258c9ef9f811d7c2ccd
  return (
    <Section>
      <Logo disabled>
        <GiAirBalloon className="hands" size={64} />
      </Logo>
      <Header>Agency</Header>
      <PTag>A Support System In Places</PTag>
      <section>
<<<<<<< HEAD
        <NavLink to="/user-form">
          <Button 
            volunteer
            id='volunteer'
            onClick={() => {
              props.setRole("volunteer");
              props.displayForms();
            }}
            >
              Volunteer
            </Button>
        </NavLink>
        <NavLink to="/user-form">
          <Button
            client
            id='client'
            onClick={() => {
              props.setRole("client");
              props.displayForms();
            }}
          >
            Client
          </Button>
        </NavLink>
=======
        <Button
          volunteer
          onClick={() => {
            props.setRole("volunteer");
            setTheRole(true);
          }}
        >
          Volunteer
        </Button>
        <Button
          client
          onClick={() => {
            props.setRole("client");
          }}
        >
          Client
        </Button>
>>>>>>> f7999fff44d6f0a4964f1258c9ef9f811d7c2ccd
      </section>
    </Section>
  );
};

export const mapDispatchToProps = dispatch => ({
  setRole: role => dispatch(setRole(role))
});

export default connect(
  null,
  mapDispatchToProps
)(LandingPage);
