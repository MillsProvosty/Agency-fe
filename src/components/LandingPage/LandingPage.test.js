import React from 'react';
import { shallow } from 'enzyme';
import { LandingPage, mapDispatchToProps } from './LandingPage';
import { setRole } from '../../actions'

describe('LandingPage', () => {
  let wrapper;

  const mockSetRole = jest.fn()
  const mockDisplayForms = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <LandingPage 
        setRole={mockSetRole}
        displayForms={mockDisplayForms}
      />
    )
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should call displayForms on click of volunteer', () => {
    wrapper.find('#volunteer').props().onClick();
    expect(mockDisplayForms).toHaveBeenCalled()
  })

  it('should call displayForms on click of client', () => {
    wrapper.find('#client').props().onClick();
    expect(mockDisplayForms).toHaveBeenCalled()
  })

  it('should set role to client on click', () => {
    wrapper.find('#client').props().onClick();
    expect(mockSetRole).toHaveBeenCalled()
  })
  
  it('should set role to volunteer on click', () => {
    wrapper.find('#volunteer').props().onClick();
    expect(mockSetRole).toHaveBeenCalled()
  })

  describe('mapDispatchToProps', () => {
    it("calls setRole with an setROle action when setRole is called", () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setRole('client');
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setRole('client');
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  })
})