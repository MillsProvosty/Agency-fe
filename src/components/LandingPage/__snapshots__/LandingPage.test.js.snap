// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should match snapshot 1`] = `
<styled.section>
  <styled.button
    disabled={true}
  >
    <GiAirBalloon
      className="hands"
      size={64}
    />
  </styled.button>
  <styled.h1>
    Agency
  </styled.h1>
  <styled.p>
    A Support System In Places
  </styled.p>
  <section>
    <styled.button
      onClick={[Function]}
      volunteer={true}
    >
      Volunteer
    </styled.button>
    <styled.button
      client={true}
      onClick={[Function]}
    >
      Client
    </styled.button>
  </section>
</styled.section>
`;
